File: 343b.py
a = raw_input()
n = len(a)
b = ""
for i in range(0,n):
    if len(b)==0 : b+=a[i]
    elif b[-1]==a[i] : b = b[:-1]
    else : b+=a[i]
if len(b)== 0 : print "Yes"
else : print "No"

File: 343b.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int
int main()
{
  string s; cin>>s;
  stack<char> v;
  for(ll i = 0;i<s.size();i++)
  {
    if(v.empty()) v.push(s[i]);
    else if(v.top()==s[i]) v.pop();
    else v.push(s[i]);
  }
  if(v.empty()) cout<<"Yes";
  else cout<<"No";
  return 0;
}
